discord-scripts/invite-management.ts

```diff
@@ -109,71 +69,75 @@ export default async function sendInvite(discordClient: Client, robot: Robot) {
           })
           await interaction.reply(
             `Here is your invite link: ${invite.url}\nThis invite expires in ${
-              maxAge / 3600
-            } hours and has a maximum of ${maxUses} uses.`,
+              maxAge / (3600 * 24)
+            } days and has a maximum of ${maxUses} uses.`,
           )
         } else {
           await interaction.reply(
             "Cannot create an invite for this type of channel.",
           )
         }
       } catch (error) {
-        await interaction.reply("An error occurred while creating an invite.")
+        await interaction.reply("An error occurred while creating the invite.")
       }
     })
 
     // Generates an invite if the channel name matches ext-*-audit format
     discordClient.on("channelCreate", async (channel) => {
-      if (channel.parent && channel.parent.name === "defense") {
-        const regex = /^ext-.*-audit$/
-        if (regex.test(channel.name)) {
-          try {
-            const channelId = channel.id
-            const maxAge = 86400 // 1 day
-            const maxUses = 5
-            const inviteUrl = await createInvite(
-              discordClient,
-              channelId,
-              maxAge,
-              maxUses,
-            )
-            if (inviteUrl) {
-              robot.logger.info(
-                `New invite created for defense audit channel: ${channel.name}, URL: ${inviteUrl}`,
-              )
-              if (channel instanceof TextChannel) {
-                channel.send(
-                  `Here is your invite link: ${inviteUrl}\n` +
-                    `This invite expires in ${
-                      maxAge / 3600
-                    } hours and has a maximum of ${maxUses} uses.`,
-                )
-              }
-            }
-            // create a new role with the exact name of the channel with permissions only to that channel
-            const role = await channel.guild.roles.create({
-              name: channel.name,
-              reason: `Role for ${channel.name} channel`,
-            })
-
-            await channel.permissionOverwrites.create(role, {
-              ViewChannel: true,
-              SendMessages: true,
-              ReadMessageHistory: true,
-            })
-            if (channel instanceof TextChannel) {
-              channel.send(
-                `Role ${role.name} created and permissions set for ${channel.name}`,
-              )
-            }
+      if (
+        channel.parent &&
+        channel.parent.name === "defense" &&
+        channel instanceof TextChannel &&
+        EXTERNAL_AUDIT_CHANNEL_REGEXP.test(channel.name)
+      ) {
+        try {
+          const channelId = channel.id
+          const maxAge = WEEK
+          const maxUses = 10
```

Comment:
At this point we've repeated our default age and uses 3 times. I think what I'd suggest here is to make `createInvite` return an object (`{ url: string; maxAge: number; maxUses: number }`) and be entirely in charge of the default age and uses (using its parameter defaults). That way we cut most of this down.

I would also consider making `createInvite` take a `TextChannel` instead of a channel id, since all callers have access to the object. That way you can also eliminate the extra channel lookup by id that we do in `createInvite`.

Reply:
I've got this all fixed now so we only have those defaults on maxAge / maxUses in the main func. Now they are set as they should be as parameter defaults, so if needed we can easily add custom values down the road. And instead of passing channel.id, it just sends the channel.object and runs invite from there.